### ISSUE TITLE
KHR_texture_transform example might not follow spec

### ISSUE THREAD

**Issue by FrankGalligan** [KhronosGroup/glTF#1724]
The spec here https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_texture_transform
says:

> Implementation Note: For maximum compatibility, it is recommended that exporters generate UV coordinate sets both with and without transforms applied, use the post-transform set in the texture texCoord field, then the pre-transform set with this extension. This way, if the extension is not supported by the consuming engine, the model still renders correctly. Including both will increase the size of the model, so if including the fallback UV set is too burdensome, either add this extension to extensionsRequired or use the same texCoord value in both places.

My understanding of this text is if the "texCoord" field is not in the extensions element, then "KHR_texture_transform" must be listed in the "extensionsRequired" element. Or am I misinterpreting the text?

If my interpretation is correct then the sample here
https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/TextureTransformTest/glTF/TextureTransformTest.gltf#L76

should have an "extensionsRequired" element as all of the KHR_texture_transform extensions do not have a texCoord set.

**Comment by bghgary** [comment]
The sample looks like this if the KHR_texture_transform is not supported:

![image](https://user-images.githubusercontent.com/19158740/70364133-1d7e2000-1840-11ea-96f2-baa379ed3181.png)

I made the extension not required so that it's obvious what happens when the extension is not supported.

**Comment by FrankGalligan** [comment]
OK I understand, it is not required so clients can tell if they render it correctly. But normally if you were to make a file like this, you would mark the extension required?

**Comment by bghgary** [comment]
If the intention is that it must have the transform for the asset to look right, yes.

**Comment by FrankGalligan** [comment]
Thanks, the only text I don't understand now is this:

>  or use the same texCoord value in both places

Can you explain further when and why you would use this? Or any links to examples? Thanks

**Comment by bghgary** [comment]
> or use the same texCoord value in both places

Yeah, I'm not sure about this either. @stevenvergenz Do you remember this?

**Comment by stevenvergenz** [comment]
That's saying to use the same accessor for both pre- and post-transform UVs. It will look wrong in one mode, but that is a fallback option.
